examples/autoqasm/5_Pulse_programming_and_dynamical_decoupling.ipynb

```diff
@@ -0,0 +1,263 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "8e867f9e",
+   "metadata": {},
+   "source": [
+    "# Pulse programming and dynamical decoupling"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e9543db0",
+   "metadata": {},
+   "source": [
+    "Native gates is a set of carefully calibrated gates directly supported by the hardware. Behind the scene, native gates themselves are implemented by analog control signals, or \"pulses\", applying on the state of the qubits. Composing a program directly with pulse operations is called \"pulse programming\", giving you even more control than native gates. For example, you can quantum operations directly in pulses in order to implement error suppression schemes such as dynamical decoupling, or error mitigation methods such zero noise extrapolation. You can also improve the elementary operation by experimenting novel custom gate implementations.\n",
+    "\n",
+    "In this notebook, we demonstrate the pulse programing features of AutoQASM with two examples, one with a qubit idling program and one with error suppression by dynamical decoupling sequences."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "e0504f72",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "import braket.experimental.autoqasm as aq\n",
+    "from braket.experimental.autoqasm import pulse\n",
+    "from braket.experimental.autoqasm.instructions import x, h, y, z, rx, rz, cz, measure\n",
+    "\n",
+    "from braket.aws import AwsDevice\n",
+    "from braket.devices import Devices\n",
+    "\n",
+    "# Rigetti Aspen-M-3 device is the targeted device for this example notebook\n",
+    "device = AwsDevice(Devices.Rigetti.AspenM3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "590f013f",
+   "metadata": {},
+   "source": [
+    "Unlike gate operations, pulse operations often target Frame objects. Basic pulse operations include setting the frequency and phase of a frame, playing a waveform, applying delay and capturing the output of a frame. The pulse operations are defined in the [pulse module](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/feature/autoqasm/src/braket/experimental/autoqasm/pulse/pulse.py) of AutoQASM. You can read this [documentation](https://docs.aws.amazon.com/braket/latest/developerguide/braket-pulse.html) to learn more about frames and waveforms.\n",
+    "\n",
+    "As a hello-world example, the pulse program below includes a `delay` instruction on the frame of the physical qubit `$0` followed by capture instruction on the same frame. The pulse program represents measuring a qubit after a variable idling duration. It can be used characterize the T1 coherence time by sampling the measurement results with different idling durations.  "
```

Comment:
I wonder if this example would be slightly more instructive if it actually showed how to measure T1 coherence time using this `idle` function? i.e., maybe `idle` should become a subroutine, and the `aq.main` program could wrap this with the necessary gates/measurements?

Reply:
Setting up a T1 experiment is a great idea. It would make the most sense being able to run the T1 program if we have it here. Currently there are still few items to iron out before the program can run on hardware. I'll look into that. In the meantime, I keep the example minimal just to demonstrate the pulse programming. I have changed the mention of T1